src/webviews/extension-side/dataviewer/dataViewerDependencyService.ts

```diff
@@ -0,0 +1,115 @@
+        }
+        return kernel;
+    }
+    private get kernelSession(): IJupyterSession {
+        if (!this.kernel.session) {
+            sendTelemetryEvent(Telemetry.NoActiveKernelSession);
+            throw new Error(DataScience.noActiveKernelSession());
+        }
+        return this.kernel.session;
+    }
+    private get packaging(): string {
+        const envType = this.kernel.kernelConnectionMetadata.interpreter?.envType;
+        const isConda = envType === EnvironmentType.Conda;
+        return isConda ? 'conda' : 'pip';
+    }
+
+    constructor(
+        @inject(IApplicationShell) private readonly applicationShell: IApplicationShell,
+        @inject(IKernelProvider) private readonly kernelProvider: IKernelProvider,
+        @inject(IsCodeSpace) private isCodeSpace: boolean
+    ) {}
+
+    public async checkAndInstallMissingDependencies(interpreter: PythonEnvironment): Promise<void> {
+        const pandasVersion = await this.getVersionOfPandas();
+
+        if (pandasVersion) {
+            if (isVersionOfPandasupported(pandasVersion)) {
+                return;
+            }
+            sendTelemetryEvent(Telemetry.PandasTooOld);
+            // Warn user that we cannot start because pandas is too old.
+            const versionStr = `${pandasVersion.major}.${pandasVersion.minor}.${pandasVersion.build}`;
+            throw new Error(DataScience.pandasTooOldForViewingFormat().format(versionStr));
+        }
+
+        sendTelemetryEvent(Telemetry.PandasNotInstalled);
+        await this.installMissingDependencies(interpreter);
+    }
+
+    private async installMissingDependencies(interpreter: PythonEnvironment): Promise<void> {
+        sendTelemetryEvent(Telemetry.PythonModuleInstall, undefined, {
+            action: 'displayed',
+            moduleName: ProductNames.get(Product.pandas)!,
+            pythonEnvType: interpreter?.envType
+        });
+        const selection = this.isCodeSpace
+            ? Common.install()
+            : await this.applicationShell.showErrorMessage(
+                  DataScience.pandasRequiredForViewing(),
+                  { modal: true },
+                  Common.install()
+              );
+
+        if (selection === Common.install()) {
+            const outputs = await executeSilently(this.kernelSession, `${this.packaging} install pandas`);
+
+            if (outputs.some((item) => item.output_type === 'error')) {
+                sendTelemetryEvent(Telemetry.FailedToInstallPandas);
+            } else {
+                sendTelemetryEvent(Telemetry.UserInstalledPandas);
+            }
+        } else {
+            sendTelemetryEvent(Telemetry.UserDidNotInstallPandas);
+            throw new Error(DataScience.pandasRequiredForViewing());
+        }
+    }
+
+    private async getVersionOfPandas(): Promise<SemVer | undefined> {
+        const outputs = await executeSilently(this.kernelSession, `import pandas; print(pandas.__version__)`);
```

Comment:
This is the bit that's not working for the "Show in Data Viewer" scenario. We are just debugging python here. What we can do here is instead of the interpreter pass the activeDebugSession. Then we can execute this same code on the debug session.

debuggerVariables.evaluate has an example of code execution.

Reply:
I think we should delerte the pandas module if we didn't import this, else we end up modifying user state by importing modules.
E.g.
```
__vsc_pandas_previously_imported = sys.modules['pandas']
try:
	import pandas as __vsc_pandas
	print(__vsc_pandas.version)
finally:
	del sys.modules['pandas']
```

Reply:
@IanMatthewHuff can we not fix that as a seprate issue, i thought thats an existing issue today.
I.e. we can focus on fixing that issue in a separate PR.

Reply:
> This change breaks the old code path for that scenario. 

Oh, if it breaks it then we have no choice but fix it here, I take back what i suggested, we cannot break something and then leave it for another PR.

> Would rather not have to do the work not, but the curr

I agree, I missed the fact that this would break existing functionality.
Feels like we are missing a test case for that, else we would have shipped a breaking change had you not caught this.

Reply:
For a while I tried to use the debugger to run the python code to check the pandas version and install it, and while it worked in the sense that I could see the execution in the debugger, I could not get the output to come back in the results.

It also drastically changed the signature of the dependency installer, from just receiving a kernel to variably receiving either a kernel or a debug session, for example.

Given that, as far as I could tell, the "Show in Data Viewer" scenario of the debugger is not implemented in the web, and with the idea of simplifying the implementation as much as possible, I have decided to keep the older dependency installer as the `.node` version of the installer, and to only use `IKernel` in the isomorphic version of the installer.

Note: At some point, Ian did suggest me this approach I ended up with, but I really wanted to see if I could make it work with the debugger. Although not being able to make it work is my fault and not the debugger's the complexity of the implementation of the dependency installer at the end felt gross, so I was also not motivated to move forward with supporting both IKernel and the debugger session in the isomorphic installer.